torchtune/models/llava_next/image_processor.py

```diff
@@ -0,0 +1,344 @@
+        image_patches = [self.center_crop(patch) for patch in image_patches]
+
+        # original image is resized, with distortion, to a square of side self.patch_size
+        # and concatenated to patches
+        patch_sized_image = F.resize(
+            image,
+            [self.patch_size, self.patch_size],
+            interpolation=self.resample,
+        ).float()
+
+        image_patches = torch.stack([patch_sized_image] + image_patches)
+
+        return image_patches
+
+
+class ImageProcessor(nn.Module):
+    def __init__(
+        self,
+        patch_size: int = 336,
+        possible_resolutions: Optional[Tuple[Tuple[int, int]]] = None,
+        max_num_chunks: int = 4,
+        resample: str = "bicubic",
+        do_rescale: bool = True,
+        rescale_factor: Union[int, float] = 1 / 255,
+        do_normalize: bool = True,
+        image_mean: Optional[Union[float, Tuple[float]]] = None,
+        image_std: Optional[Union[float, Tuple[float]]] = None,
+    ) -> None:
+        """
+        Args:
+            patch_size (int): Size of the patches to divide the image into.
+            possible_resolutions (Optional[List[Tuple[int, int]]]): List of possible resolutions as tuples (height, width).
+            max_num_chunks (int): Only used possible_resolutions is NOT given. Maximum number of chunks for processing
+            high-resolution images.
+            resample (str): Resampling method used when resizing images.
+            do_rescale (bool): Flag to determine whether to rescale the image by `rescale_factor`.
+            rescale_factor (Union[int, float]): Scale factor used if rescaling the image.
+            do_normalize (bool): Flag to determine whether to normalize the image.
+            image_mean (Optional[Union[float, List[float]]]): Mean values for normalization.
+            image_std (Optional[Union[float, List[float]]]): Standard deviation values for normalization.
+        """
+        super().__init__()
+
+        logger.info("Initializating ImageProcessor...")
+
+        if image_mean is None:
+            image_mean = [0.48145466, 0.4578275, 0.40821073]
+        if image_std is None:
+            image_std = [0.26862954, 0.26130258, 0.27577711]
+
+        # If possible_resolutions are not given, then calculate possible ones based on max_num_chunks
+        if not possible_resolutions:
+            possible_resolutions = find_supported_resolutions(
+                max_num_chunks=max_num_chunks, patch_size=patch_size
+            )
+        else:
+            possible_resolutions = possible_resolutions
+
+        logger.info(f"possible_resolutions: {possible_resolutions}")
+
+        self.patchfy = GetImagePatches(
+            possible_resolutions=possible_resolutions,
+            patch_size=patch_size,
+            resample=resample,
+        )
+
+        _preprocess = []
+        if do_rescale:
+            _preprocess.append(v2.Normalize(mean=[0] * 3, std=[1 / rescale_factor] * 3))
```

Comment:
Good question! I think that people usually just do image = image / rescaling_factor, but since I am using nn.sequential, I wanted to leverage that to avoid creating a new function of adding more code to forward().

Let me think a bit about it.